doc/source/train/examples/lightning/lightning_exp_tracking.ipynb

```diff
@@ -0,0 +1,498 @@
+   "source": [
+    "# Define a dummy model\n",
+    "class DummyModel(pl.LightningModule):\n",
+    "    def __init__(self):\n",
+    "        super().__init__()\n",
+    "        self.layer = torch.nn.Linear(3, 1)\n",
+    "\n",
+    "    def forward(self, x):\n",
+    "        return self.layer(x)\n",
+    "\n",
+    "    def training_step(self, batch, batch_idx):\n",
+    "        x, y = batch\n",
+    "        y_hat = self(x)\n",
+    "        loss = F.binary_cross_entropy_with_logits(y_hat.flatten(), y.float())\n",
+    "\n",
+    "        # The metrics below will be reported to Loggers\n",
+    "        self.log(\"train_loss\", loss)\n",
+    "        self.log_dict({\"metric_1\": 1 / (batch_idx + 1), \"metric_2\": batch_idx * 100})\n",
+    "        return loss\n",
+    "\n",
+    "    def configure_optimizers(self):\n",
+    "        return torch.optim.Adam(self.parameters(), lr=1e-3)"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Define your loggers\n",
+    "\n",
+    "For offline loggers, no changes are required in the Logger initialization.\n",
+    "\n",
+    "For online loggers (W&B and CometML), you need to do two things:\n",
+    "- Set up your API keys as environment variables.\n",
+    "- Set `rank_zero_only.rank = None` to avoid Lightning creating a new experiment run on the driver node. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 29,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "CometLogger will be initialized in online mode\n"
+     ]
+    }
+   ],
+   "source": [
+    "from pytorch_lightning.loggers.wandb import WandbLogger\n",
+    "from pytorch_lightning.loggers.comet import CometLogger\n",
+    "from pytorch_lightning.loggers.mlflow import MLFlowLogger\n",
+    "from pytorch_lightning.loggers.tensorboard import TensorBoardLogger\n",
+    "from pytorch_lightning.utilities.rank_zero import rank_zero_only\n",
+    "import wandb\n",
+    "\n",
+    "\n",
+    "# A callback to login wandb in each worker\n",
+    "class WandbLoginCallback(pl.Callback):\n",
+    "    def __init__(self, key):\n",
+    "        self.key = key\n",
+    "\n",
+    "    def setup(self, trainer, pl_module, stage) -> None:\n",
+    "        wandb.login(key=self.key)\n",
+    "\n",
+    "\n",
+    "def create_loggers(name, project_name, save_dir=\"./logs\", offline=False):\n",
+    "    # Avoid creating a new experiment run on the driver node.\n",
+    "    rank_zero_only.rank = None\n",
+    "\n",
+    "    # Wandb\n",
+    "    wandb_api_key = os.environ.get(\"WANDB_API_KEY\", None)\n",
+    "    wandb_logger = WandbLogger(\n",
+    "        name=name, \n",
+    "        project=project_name, \n",
+    "        save_dir=f\"{save_dir}/wandb\", \n",
+    "        offline=offline\n",
+    "    )\n",
+    "    callbacks = [] if offline else [WandbLoginCallback(key=wandb_api_key)]\n",
```

Comment:
Hmm, interesting. Is this the recommended way of doing wandb auth? What if we just make the API key env var set on all workers?

Reply:
Yeah if we already set the API key, then we don't have to create this callback.